underdocs-error/src/main/kotlin/underdocs/error/pretty/DefaultErrorPrettyPrinter.kt

```diff
@@ -1,7 +1,30 @@
 package underdocs.error.pretty
 
+import underdocs.error.BaseUnderdocsError
+
 class DefaultErrorPrettyPrinter : ErrorPrettyPrinter {
-    override fun print(exception: Exception): String {
-        TODO("Not yet implemented")
+    override fun print(exception: Exception) = if (exception is BaseUnderdocsError) {
+        printBaseUnderdocsException(exception)
+    } else {
+        printUnexpectedException(exception)
     }
+
+    private fun printBaseUnderdocsException(exception: BaseUnderdocsError) = """
+        |${exception.title.toUpperCase()} - Error Code ${exception.code}
+        |Details
+        |${exception.getDescription()}
+        |
+        |Possible solution
+        |${exception.getSolution()}
+        |${exception.cause.let { if (it?.message != null) "Error cause: ${it.message}" else "" }}
```

Comment:
@battila7 
I hope this is what you meant by the scope function comment, but this one still leaves a blank line, if the cause is null. Is it fine or would this need some more refactoring?

Reply:
I was thinking more like:

~~~~Kotlin
var prettyPrinted = "something"

exception.cause?.apply {
  prettyPrinted += "Error cause: $message"
}
~~~~

Thus, keeping the original control flow, but replacing the `if` with a scope function.

However, now it got me thinking. Maybe including the message of the cause is not even necessary after all, what do you think? The description is expected to provide enough context for the user regarding the error, therefore, in my opinion, the message from the cause (which is something of a lower-level stuff) would not help that much.

Reply:
You are right, this might be a bit better. 

I was thinking about if the message is necessary or not too. If the descriptions are expressive/extensive enough, it would probably have enough context. So in my opinion it's not that necessary.

Reply:
Cool! Feel free to remove it then 👍